tests/test_isPrimaryFlag.py

```diff
@@ -36,31 +153,126 @@ def setUp(self):
+        wcs = self.exposure.getWcs()
+        tractBBox = Box2I(Point2I(100, 100), Extent2I(900, 900))
+        skyMap = MockSkyMap([tractBBox], wcs, 3)
+        tractInfo = skyMap[0]
+        patchInfo = tractInfo[0, 0]
+        patchBBox = patchInfo.getInnerBBox()
+
+        schema = SourceCatalog.Table.makeMinimalSchema()
+        # Initialize the detection task
+        detectionTask = SourceDetectionTask(schema=schema)
+
+        # Initialize the fake source injection task
+        skyConfig = SkyObjectsTask.ConfigClass()
+        skySourcesTask = SkyObjectsTask(name="skySources", config=skyConfig)
+        schema.addField("merge_peak_sky", type="Flag")
+
+        # Initialize the deblender task
+        scarletConfig = ScarletDeblendTask.ConfigClass()
+        scarletConfig.maxIter = 20
+        scarletConfig.columnInheritance["merge_peak_sky"] = "merge_peak_sky"
+        deblendTask = ScarletDeblendTask(schema=schema, config=scarletConfig)
+
+        # We'll customize the configuration of measurement to just run the
+        # minimal number of plugins to make setPrimaryFlags work.
+        measureConfig = SingleFrameMeasurementTask.ConfigClass()
+        measureConfig.plugins.names = ["base_SdssCentroid", "base_SkyCoord"]
+        measureConfig.slots.psfFlux = None
+        measureConfig.slots.apFlux = None
+        measureConfig.slots.shape = None
+        measureConfig.slots.modelFlux = None
+        measureConfig.slots.calibFlux = None
+        measureConfig.slots.gaussianFlux = None
+        measureTask = SingleFrameMeasurementTask(config=measureConfig, schema=schema)
+        primaryConfig = SetPrimaryFlagsTask.ConfigClass()
+        setPrimaryTask = SetPrimaryFlagsTask(config=primaryConfig, schema=schema,
+                                             name="setPrimaryFlags", isSingleFrame=False)
+
+        table = SourceCatalog.Table.make(schema)
+        # detect sources
+        detectionResult = detectionTask.run(table, coadds["test"])
+        catalog = detectionResult.sources
+        # add fake sources
+        skySources = skySourcesTask.run(mask=self.exposure.mask, seed=0)
+        for foot in skySources[:5]:
+            src = catalog.addNew()
+            src.setFootprint(foot)
+            src.set("merge_peak_sky", True)
+        # deblend
+        result = deblendTask.run(coadds, catalog)
+        # measure
+        measureTask.run(result["test"], self.exposure)
+        outputCat = result["test"]
+        # Set the primary flags
+        setPrimaryTask.run(outputCat, skyMap=skyMap, tractInfo=tractInfo, patchInfo=patchInfo)
+
+        # There should be the same number of deblenedPrimary and
+        # deblendedModelPrimary sources,
+        # since they both have the same blended sources and only differ
+        # over which model to use for the isolated sources.
+        self.assertEqual(
+            np.sum(outputCat["detect_isDeblendedPrimary"]),
+            np.sum(outputCat["detect_isDeblendedModelPrimary"]))
```

Comment:
I'll add a check that they are non-zero, but checking against a known number is not advisable. As you know so many things upstream affect the creation of footprints and peaks that the number of peaks and footprints changes with practically every weekly, causing this test to trip frequently when people make changes upstream.

Reply:
FYI: this is adjusted in DM-27929 because they will be slightly different now that meas_extensions_scarlet skips sky sources, but in a way that can be predicted and tested.